core/domain/html_validation_service.py

```diff
@@ -877,15 +880,40 @@ def get_filename_with_dimensions(old_filename, exp_id):
     return new_filename
 
 
-def get_invalid_svg_tags(svg_string):
-    """Returns a set of all invalid tags for the provided SVG.
+def get_invalid_svg_tags_and_attrs(svg_string):
+    """Returns a set of all invalid tags and attributes for the provided SVG.
 
     Args:
         svg_string: str. The SVG string.
 
     Returns:
-        set. A set of invalid tags.
+        tuple. A 2-tuple, the first element of which is a list of invalid tags,
+        and the second element of which is a list of invalid attributes.
     """
     soup = bs4.BeautifulSoup(svg_string.encode('utf-8'), 'html.parser')
-    return set([tag.name for tag in soup.find_all()]).difference(
-        feconf.ALLOWED_SVG_TAGS)
+    invalid_elements = []
+    invalid_attrs = []
+    for element in soup.find_all():
+        if element.name.lower() in constants.ALLOWED_SVG_TAGS:
+            for attr in element.attrs:
+                if attr.lower() not in constants.ALLOWED_SVG_ATTRS:
+                    invalid_attrs.append(attr)
+        else:
+            invalid_elements.append(element.name)
+    return (invalid_elements, invalid_attrs)
+
+
+def is_xml_parsable(xml_string):
```

Comment:
I don't see any tests for this function -- please always add comprehensive tests for new functions before sending code for review.